feat(networking): make record store r/w non blocking

joshuef opened the pull request:
## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 19 Jul 23 10:40 UTC
This pull request includes two patches. 

The first patch makes the record store read/write operations non-blocking. It also adds a TODO comment about cleaning up records if an insert fails. Additionally, it spawns a tokio task to asynchronously write records to disk and handle errors.

The second patch removes an unused function called `add_to_routing_table` in the networking module of the codebase. The function is no longer needed and has been deleted.
<!-- reviewpad:summarize:end -->

joshuef committed:
feat(networking): make record store r/w non blocking

RolandSherwin commented on sn_networking/src/lib.rs:
```diff
@@ -487,17 +487,6 @@ impl Network {
-        })?;
-        receiver.await?
-    }
-
```
If we are not using the function, then I think we can get rid of the corresponding `SwarmCmd` as well?

joshuef committed:
chore(networking): remove PutRecord kad query tracking.

This doesn't tell us anything useful and is more state
to keep track of in the swarm driver

joshuef committed:
test(networking): add read loop to await async disk write

verified puts write to disk in async manner now, returning immediately.
So a  straight afterwards is not guaranteed to succeed. Thus we
add a small but finate get loop before we assert

joshuef committed:
chore(cli): verify chunks when uploading

joshuef commented:
atop https://github.com/maidsafe/safe_network/pull/543

joshuef committed:
feat(networking): add in verify_store to kad::put apis

joshuef committed:
chore: remove useless log

joshuef committed:
ci: increase dbc spend test times now we validate each spend

joshuef committed:
chore(networking): reset something_was_found during verification loop.

Other small fixes

joshuef committed:
test: update dbc verify test to rely on  api alone.

This api now validates the store, which should fail due to the
corrupt DBC

joshuef committed:
chore(node): add spend_no_verify fn and use in some tests
